pl_examples/hydra_examples/pl_template.py

```diff
@@ -45,7 +46,29 @@ def main(cfg: DictConfig):
     # ------------------------
     # 2 INIT TRAINER
     # ------------------------
-    trainer = Trainer(**init_trainer(cfg))
+    def conf_init(cfg, key):
+        return hydra.utils.instantiate(getattr(cfg, key)) if hasattr(cfg, key) else None
+
+    profiler = conf_init(cfg.pl, "profiler")
+    early_stopping = conf_init(cfg.pl, "early_stopping")
+    checkpoint = conf_init(cfg.pl, "checkpoint")
+    logger = conf_init(cfg.pl, "logger")
+    if cfg.callbacks_list:
+        callbacks = [hydra.utils.instantiate(callback) for callback in cfg.callbacks_list]
+    else:
+        callbacks = []
+    trainer = Trainer(
+        **asdict(
+            TrainerIn(
+                **cfg.pl.trainer,
+                logger=logger,
+                profiler=profiler,
+                checkpoint_callback=checkpoint,
+                early_stop_callback=early_stopping,
+                callbacks=callbacks,
+            )
+        )
```

Comment:
Okay, now I understand why it works.

Wouldn't this work too?
```python
trainer = Trainer(
  **cfg.pl.trainer,
  logger=logger,
  profiler=profiler,
  checkpoint_callback=checkpoint,
  early_stop_callback=early_stopping,
  callbacks=callbacks
)
```

Reply:
It does!